### ISSUE TITLE
OVH fails

### ISSUE THREAD

**Issue by duythongle** [kubicorn/kubicorn#632]
I followed OVH walkthrough and got an error while applying:
```
thongld@i3wattos  ~  source openrc.sh
thongld@i3wattos  ~  kubicorn apply myfirstk8s
2018-04-24T03:38:17-07:00 [ℹ]  Selected [fs] state store
2018-04-24T03:38:17-07:00 [ℹ]  Loaded cluster: myfirstk8s
2018-04-24T03:38:17-07:00 [ℹ]  Init Cluster
2018-04-24T03:38:19-07:00 [ℹ]  Query existing resources
2018-04-24T03:38:23-07:00 [ℹ]  Resolving expected resources
2018-04-24T03:38:23-07:00 [ℹ]  Reconciling
2018-04-24T03:38:25-07:00 [✔]  Created Network [1a580eb0-3418-44ff-8a6e-a122837a392e]
2018-04-24T03:38:29-07:00 [✔]  Created Subnet [cbed8ecd-4257-481c-80fb-5967d0080537]
2018-04-24T03:38:35-07:00 [✔]  Created SecurityGroup [f3c7fe35-ea36-4090-8c27-68280b12f010]
2018-04-24T03:38:35-07:00 [ℹ]  Parsing bootstrap script from GitHub [https://raw.githubusercontent.com/kubicorn/bootstrap/master/ovh_k8s_ubuntu_16.04_master.sh]
2018-04-24T03:38:44-07:00 [✖]  Error during apply of atomic reconciler, attempting clawback: Expected HTTP response code [201 202] when accessing [POST https://compute.bhs1.cloud.ovh.net/v2/d5430f3c1ea84d198d7a91bcb224dbc0/servers], but got 400 instead
{"badRequest": {"message": "Invalid key_name provided.", "code": 400}}
2018-04-24T03:38:44-07:00 [!]
2018-04-24T03:38:44-07:00 [!]  Attempting to backtrack and cleanup created resources.
2018-04-24T03:38:44-07:00 [!]
2018-04-24T03:38:45-07:00 [✔]  Deleted SecurityGroup [f3c7fe35-ea36-4090-8c27-68280b12f010]
2018-04-24T03:38:46-07:00 [✔]  Deleted Subnet [cbed8ecd-4257-481c-80fb-5967d0080537]
2018-04-24T03:38:48-07:00 [✔]  Deleted Network [1a580eb0-3418-44ff-8a6e-a122837a392e]
2018-04-24T03:38:48-07:00 [✖]  Unable to commit state store: Nil cluster spec
```

**Comment by abdelhalim2012** [comment]
I encounter the same issue..

**Comment by xlucas** [comment]
I'm currently working on this issue, changes introduced to the provider broke the key name resolution.

**Comment by xlucas** [comment]
Related to #590

**Comment by xmudrii** [comment]
Thanks for reporting the issue! We believe this issues should be fixed by #639.

As this is duplicate of #590, and it contains some more details, I'm going to close this issue. If you have any questions, let us know!
